providers/providers.go

```diff
@@ -163,7 +163,11 @@ func readTimeValue(i interface{}) (time.Time, error) {
 func (pm *ProviderManager) addProv(k *cid.Cid, p peer.ID) error {
 	iprovs, ok := pm.providers.Get(k.KeyString())
 	if !ok {
-		iprovs = newProviderSet()
+		stored, err := loadProvSet(pm.dstore, k)
+		if err != nil {
+			return err
+		}
+		iprovs = stored
```

Comment:
> Cache miss might mean that provider records where offloaded to datastore and need to be brough back up in order to have a consistent cache entry.

I *think* this makes sense, but what would does inconsistency mean in the context of provider records?